### PR TITLE
Final msvc build image i mean it i hope

### PR DESCRIPTION
**Pull request by myk002**
- revert to the old version of protobuf for native build. I just cannot get the linking errors with the newer version figured out : (
- add ccache back in, and patch it to work with msvc
- build a newer version of git so the image can be used more easily as a personal build environment
- fix the dfhack-* build scripts so they can be run repeatedly
- organize and reorder the steps in the dockerfile so (future) rebuilds should go quicker (i.e. steps that are more likely to change now appear later in the file)

### CODE REVIEW

File: msvc/dfhack-test [hunk 1]
```diff
@@ -1,21 +1,11 @@
-#!/bin/bash -e
+#!/bin/bash
```
Comment (BenLubar) [inline msvc/dfhack-test:1]: removing -e means we won't get exit statuses from commands that fail

you might want to add `status=$?` after the important command and then `exit $status` at the end

Comment (myk002) [reply msvc/dfhack-test:1]: Ah, ok. Sorry, I'll fix that. The bash man page didn't really explain what it was for (just that it was inherited from the parent shell) and it seemed to cause the script to have no effect after the first time I ran it. I'll add in a status return when I get the chance

Comment (BenLubar) [reply msvc/dfhack-test:1]: `-e` means that if any command returns non-zero, the bash shell immediately exits with that code

Comment (myk002) [reply msvc/dfhack-test:1]: Ok, I think I understand now. I bet one of the wine commands returns zero the first time it is run and non-zero thereafter. It was very strange to have the scripts returning immediately with zero output.

**Pull request merged**